controllers/openstackmachine_controller.go

```diff
@@ -359,29 +359,37 @@ func (r *OpenStackMachineReconciler) reconcileNormal(ctx context.Context, scope
 		return ctrl.Result{RequeueAfter: waitForInstanceBecomeActiveToReconcile}, nil
 	}
 
-	if util.IsControlPlaneMachine(machine) {
-		if openStackCluster.Spec.APIServerLoadBalancer.Enabled {
-			err = r.reconcileLoadBalancerMember(scope, openStackCluster, machine, openStackMachine, instanceNS, clusterName)
-			if err != nil {
-				handleUpdateMachineError(scope.Logger, openStackMachine, errors.Errorf("LoadBalancerMember cannot be reconciled: %v", err))
-				return ctrl.Result{}, nil
-			}
-		} else if !openStackCluster.Spec.DisableAPIServerFloatingIP {
-			floatingIPAddress := openStackCluster.Spec.ControlPlaneEndpoint.Host
-			if openStackCluster.Spec.APIServerFloatingIP != "" {
-				floatingIPAddress = openStackCluster.Spec.APIServerFloatingIP
-			}
-			fp, err := networkingService.GetOrCreateFloatingIP(openStackMachine, openStackCluster, clusterName, floatingIPAddress)
-			if err != nil {
-				handleUpdateMachineError(scope.Logger, openStackMachine, errors.Errorf("Floating IP cannot be got or created: %v", err))
-				return ctrl.Result{}, nil
-			}
-			port, err := computeService.GetManagementPort(openStackCluster, instanceStatus)
-			if err != nil {
-				err = errors.Errorf("getting management port for control plane machine %s: %v", machine.Name, err)
-				handleUpdateMachineError(scope.Logger, openStackMachine, err)
-				return ctrl.Result{}, nil
-			}
+	if !util.IsControlPlaneMachine(machine) {
```

Comment:
Should we really stop for non-control-plane machines this early?
Another idea could be to create a method `reconcileControlPlaneFloatingIP` and do sth like (untested, not written in an IDE 🙂)

```go
	if util.IsControlPlaneMachine(machine) {
		if err := r.reconcileControlPlaneFloatingIP(scope, openStackCluster, machine, openStackMachine, instanceNS, clusterName) {
			return ctrl.Result{}, err
		}
	}
	scope.Logger.Info("Reconciled Machine create successfully")
	return ctrl.Result{}, nil
```

WDYT?

Reply:
I didn't change the logic here :) if you compare with orignial code 

we can talk about here more :) 

and yes, I think we can refactor the code to a function~

Reply:
Or this

```go
if util.IsControlPlaneMachine(machine) {
scope.Logger.Info("Reconciled Machine create and floating IP successfully")
	return ctrl.Result{}, r.reconcileControlPlaneFloatingIP(scope, openStackCluster, machine, openStackMachine, instanceNS, clusterName)
}
scope.Logger.Info("Reconciled Machine create successfully")
return ctrl.Result{}, nil
```

Reply:
> I didn't change the logic here :) if you compare with orignial code

With "this early" I mean that all the code underneath will never be called if it's a worker machine - even not the last `return`. This can be tricky to understand in the future.

Reply:
But I'm totally fine to do this refactoring in a follow-up PR 👍🏻

Reply:
yeah, I think it make sense to follow up 
as we focus on the logic /doc change this PR :)